File: js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

function linearSearch(valueToFind, arrayToSearchThrough) {
    let result = []
    for(let i = 0; i < arrayToSearchThrough.length; i++) {
        if(arrayToSearchThrough[i] === valueToFind) {
            result.push(i)
        }
    }
    console.log(result)
};

linearSearch("n", ["b", "a", "n", "a", "n", "a", "s"])

File: python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

#release 0
# def linear_search(value_to_find, array_to_search_through):
#     count = -1
#     for i in array_to_search_through:
#         count += 1;
#         if i == value_to_find:
#             return count
    
#     return None

#release 1
def global_linear_search(value_to_find, array_to_search_through):
    occur = []
    count = -1
    for i in array_to_search_through:
        count += 1;
        if i == value_to_find:
            occur.append(count)

    if(len(occur) == 0):
        return None
    
    return occur

